tests/databases/utilities.py

```diff
@@ -0,0 +1,130 @@
+def assert_database_state(state1, state2):
+    assert state1 == state2
+
+
+class HashableDict(dict):
+    def __hash__(self):
+        return hash((frozenset(self), frozenset(self.values())))
+
+    def __eq__(self, other):
+        return super().__eq__(other)
+
+
+def to_hashable(items):
+    for key, value in items:
+        if key == 'a':
+            yield key, tuple(tuple(v) for v in value)
+
+        elif key == 'b':
+            yield (
+                key,
+                tuple(
+                    (atom1, atom2, order, tuple(periodicity))
+                    for atom1, atom2, order, periodicity in value
+                )
+            )
+
+        elif key == 'm':
+            yield key, tuple(tuple(v) for v in value)
+
+        elif key == 'BB':
+            yield key, tuple(map(HashableDict, value))
+
+        elif key == 'aI':
+            yield key, tuple(tuple(v) for v in value)
+
+        elif key == 'bI':
+            yield key, tuple(tuple(v) for v in value)
+
+        elif key == 'nBB':
+            yield key, tuple(value)
+
+        else:
+            yield key, value
+
+
+def get_entry(item):
+    """
+    Create a :class:`.DatabaseEntry` from a :class:`dict`.
+
+    Parameters
+    ----------
+    item : :class:`dict`
+        A record retrieved from a MongoDB collection.
+
+    Returns
+    -------
+    :class:`.DatabaseEntry`
+        The entry created from `item`.
+
+    """
+
+    return DatabaseEntry(**{
+        key: value
+        for key, value in to_hashable(item.items()) if key != '_id'
+    })
+
+
+class DatabaseEntry:
+    """
+    Represents an record in a MongoDB collection.
```

Comment:
a record?

Reply:
Synonym for entry. Will use entry for consistency.